### PR TITLE
Add ELF virtual address loading

### LINKED ISSUE #6: Implement correct ELF loading at address specified in header

### PR DESCRIPTION
**Pull request by nerett**
Load executable segments to memory by their virtual addresses

Close #6

### CODE REVIEW

File: main.cpp
Comment (vian96) [inline main.cpp:1]: changes seem unrelated. this file is changed in another commit so it will cause conflict. please checkout this file so it does not change this file if not necessary

**Comment by vian96** [comment]
please remove changes in main so we can merge it

File: main.cpp
Comment (nerett) [reply main.cpp:1]: done

File: elf_reader.hpp [hunk 1]
```diff
@@ -7,57 +7,85 @@
...
             }
 
-            char* data = nullptr;
-            uint64_t segment_start = 0, segment_end = 0;
             for (const auto& segment : reader.segments) {
-                if (segment->get_type() == ELFIO::PT_LOAD && (segment->get_flags() & ELFIO::PF_X)) {
-                    segment_start = segment->get_virtual_address();
-                    segment_end = segment_start + segment->get_memory_size();
+                if ((segment->get_type() == ELFIO::PT_LOAD) &&
+                    (segment->get_flags() & ELFIO::PF_X)) {
```
Comment (vian96) [inline elf_reader.hpp:32]: does that mean we only load executable pages? if so, why?

**Comment by vian96** [comment]
Other than question about executable segments LGTM, ping me so I approve

File: elf_reader.hpp (hunk 1 above)
Comment (nerett) [reply elf_reader.hpp]: fixed

**Comment by vian96** [comment]
LGTM

**Review by vian96 — APPROVED** [review]
